docs/components/compact.md

```diff
@@ -9,37 +9,160 @@ menu: components
 The `thanos compact` command applies the compaction procedure of the Prometheus 2.0 storage engine to block data stored in object storage.
 It is generally not semantically concurrency safe and must be deployed as a singleton against a bucket.
 
-It is also responsible for downsampling of data:
+Compactor is also responsible for downsampling of data:
 
-- creating 5m downsampling for blocks larger than **40 hours** (2d, 2w)
-- creating 1h downsampling for blocks larger than **10 days** (2w).
+* Creating 5m downsampling for blocks larger than **40 hours** (2d, 2w)
+* Creating 1h downsampling for blocks larger than **10 days** (2w)
 
 Example:
 
 ```bash
 thanos compact --data-dir /tmp/thanos-compact --objstore.config-file=bucket.yml
 ```
 
-The content of `bucket.yml`:
+Example content of `bucket.yml`:
 
 ```yaml
 type: GCS
 config:
   bucket: example-bucket
 ```
 
-The compactor needs local disk space to store intermediate data for its processing. Generally, about 100GB are recommended for it to keep working as the compacted time ranges grow over time.
-On-disk data is safe to delete between restarts and should be the first attempt to get crash-looping compactors unstuck.
+**Compactor is the only Thanos component which should have write access to object storage**
+
+> **NOTE:** High availability for Compactor is generally not required. See [Availability](#availability) section.
+
+## Warning: Only One Singleton has to run against single Object Storage.
+
+Because there is no safe locking mechanism for all object storage provides, currently you need to on your own ensure that only
+single Compactor is running against single bucket. Running more can result with [Overlap Issues](../operating/troubleshooting.md#overlaps)
+that has to be resolved manually.
+
+Singleton, means also that there should be no local compaction on Prometheus side that could potentially cause both compacted and non compacted blocks
+being uploaded by sidecar. This is why uploaded flag is still under a separate `--shipper.upload-compacted` flag that helps to ensure that compacted blocks are uploaded before anything else.
+The singleton rule is also why local Prometheus compaction has to be disabled in order to use sidecar with upload option. Use hidden `--shipper.ignore-unequal-block-size` to override this check (on your own risk).
+
+> **NOTE:** In further Thanos version it's possible that both restrictions will be removed with production status of [vertical compaction](#vertical-compactions) which is worked on.
+
+The only acceptable case of running multiple Compactors against single Bucket is when you want to [scale compaction process](#scalability).
```

Comment:
It seems a little odd, that we first write essentially "never run more than one compactor", and then end with "except you want scale". We should either not say this at all here or elaborate further.